src/hooks/use-collaborator-activity-notifications.ts

```diff
@@ -0,0 +1,103 @@
+import { dispatch, useSelect } from '@wordpress/data';
+import { useEffect, useRef } from '@wordpress/element';
+import { store as noticesStore } from '@wordpress/notices';
+
+import {
+	AwarenessStoreSelectors,
+	UserState,
+	store as awarenessStore,
+} from '@/store/awareness-store';
+import { SettingsStoreSelectors } from '@/store/settings-store';
+
+/**
+ * Custom hook to show notifications when users join or leave the post.
+ * Notifications are shown based on the settings configured by the user.
+ */
+export function useCollaboratorActivityNotifications() {
```

Comment:
My logic for calling this hook `CollaboratorActivity` is that, this can be the single area where all notifications influenced by a collaborator's activity could go.

That's why the logic for actually sending notifications when a user joins/leaves is under presence notifications. It's a presence change related to collaborator activity.